tests/mmptest/src/BindingProjectTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+
+namespace Xamarin.MMP.Tests
+{
+	public class BindingProjectTests
+	{
+		static string BindingName (bool full) => full ? "XM45Binding" : "MobileBinding";
+		
+		static void BuildLinkedTestProjects (string tmpDir, bool full = false, bool removeTFI = false)
+		{
+			string bindingName = BindingName (full);
+			TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) {
+				ProjectName = bindingName + ".csproj",
+				ItemGroup = MMPTests.CreateSingleNativeRef (Path.GetFullPath (MMPTests.SimpleDylibPath), "Dynamic"),
+				StructsAndEnumsConfig = "public class UnifiedWithDepNativeRefLibTestClass {}"
+			};
+
+			if (removeTFI)
+				test.CustomProjectReplacement = new Tuple<string, string> (@"<TargetFrameworkVersion>v4.5</TargetFrameworkVersion>", "");
+
+			string projectPath = TI.GenerateBindingLibraryProject (test);
+			TI.BuildProject (projectPath, true);
+
+			string referenceCode = string.Format (@"<Reference Include=""{0}""><HintPath>{1}</HintPath></Reference>", bindingName, Path.Combine (tmpDir, "bin/Debug", bindingName + ".dll"));
+
+			test = new TI.UnifiedTestConfig (tmpDir) {
+				References = referenceCode,
+				TestCode = "System.Console.WriteLine (typeof (ExampleBinding.UnifiedWithDepNativeRefLibTestClass));",
+				XM45 = full					
+			};
+			TI.TestUnifiedExecutable (test);
+		}
+
+		[Test]
+		public void ShouldRemovePackagedLibrary_OnceInBundle ()
```

Comment:
This test was actually broken in master since it lacked the [Test] attribute, which was removed by me in a poor merge (https://github.com/xamarin/xamarin-macios/commit/d4e32a96f21e18dc7e93635e81df35664132ed0d).